upload/common/client_side_checksum_handler.py

```diff
@@ -0,0 +1,69 @@
+import os
+import sys
+import time
+
+from dcplib.checksumming_io import ChecksummingSink
+from dcplib.s3_multipart import get_s3_multipart_chunk_size
+
+# Checksum(s) to compute for file; current options: crc32c, sha1, sha256, s3_etag
+CHECKSUM_NAMES = ['crc32c']
+
+
+class ClientSideChecksumHandler:
+    """ The ClientSideChecksumHandler takes in a file as a parameter and handles any behavior related to
+    check-summing the file
+    on the client-side, returning a tag that can be used as metadata when the file is uploaded to S3."""
+
+    def __init__(self, filename=None, data=None):
+        self._checksums = {}
+        self._filename = filename
+        self._data = data
+
+    def get_checksum_metadata_tag(self):
+        """ Returns a map of checksum values by the name of the hashing function that produced it."""
+        if not self._checksums:
+            print("Warning: No checksums have been computed for this file.")
+        return {str(_hash_name): str(_hash_value) for _hash_name, _hash_value in self._checksums.items()}
+
+    def compute_checksum(self):
```

Comment:
This function/class seems complicated for what its actually doing on the server side. I don't think there is any need to add "file" param support at this stage which would get rid of a lot of these if statements? As far as I can tell, this class only needs to compute this checksum on the content passed in from UploadArea.store_file()

Reply:
Filename is more just forward support when we don't actually want to pass around the full data that needs to be checksummed (i.e. with larger files). I'd like to add s3 support soon in which case the filename will be needed instead of passing the data after opening the s3 file if that makes sense.